PaddlePaddle/Paddle
# Update from offical repo (#1)

* fix bug of 'shuf' and 'gshuf' in demo/recommendation/preprocess.sh

* Remove those extra trailing empty lines.

* Rephrase the first paragraph

* remove random flag

* modify format of cluster_train_cn.md

* add link of demo/recommendation

* fix dead links and refine tutorials/index

* modify markdown format of cluster_train_en.md and add link of demo/recommendation

* remove gpu memory alloc

* modify details

* add refer link of Kubernetes in cluster_train_cn.md

* add refer link of Kubernetes in cluster_train_en.md

* Clean BatchNorm Code.

* add Layer::createFunction

* fix dead links on quick_start

* travis for check broken links

* Added support for cudnn v6 and cuda 8.0

* Matrix API refactor, when passing parameters, convert shared_ptr (MatrixPtr) to
reference or raw matrix (Matrix & or Matrix *)
contextProjectionForward
contextProjectionBackward
contextProjectionBackwardData
contextProjectionBackwardWeight
classificationError
The mul functions would be updated later.

* Refactor MUL functions, pass object reference instead of shared_ptr.

* set -e for docs.sh

* update docker_install_en.rst

* Add priorbox layer gpu unit test.

* Add excluded_chunk_types to ChunkEvaluator

The chunks of types in excluded_chunk_types will not be counted in
ChunkEvaluator. This is useful for tasks such as SRL, in which chunks of
type V (verb) will not be taken into account in evaluation.

* move TEST from test_matrixCompare.cpp to cross_map_normal_op_test.cpp

* Refine Code

* fix style issues

* Add const in ParameterUpdater init

* follow comments

* Fix bug in config_parse.py when batch_norm layer is used in RecurrentLayerGroup.

* change float to real in two test

Change float in test_ConvTrans and test_ConvUnify to real.

* Start doing mnist_train_api

* Remove not used params in GradientMachine::start

* Add GradientMachine::start/finish to API

* Remove unused cost parameter in ParameterUpdater

* Adding namespace in timing macros

Sometime those macros are used under different namespaces. We need to use namespace ::paddle
to make it compile correctly.

Change-Id: I57a6d6ec8cd0d680b584aab62d72a35c226a24a4

* close log info in BN.

* fix array style problem

* refine docs.sh

* Change type float to real.

* Support user specified label input in tests

* refine docs.sh

* change type to bool.

* 按照评审人的意见进行了修改

* Update index_cn.md

* modify

* modify load_missing_parameter_strategy

* modify the url reference

correct the original to "../text_generation/index_cn.md"

* Wait for reading data.

* A tiny fix in PyDataProvider2

* hidden decorator kwargs in DataProvider.__init__
* also add unit test for this.

* Try to read data in mnist

* Use numpy in DenseScanner.

* Try expose ParamUpdater::update

* A simplest train file for mnist added.

* A tiny fix in PyDataProvider2

* hidden decorator kwargs in DataProvider.__init__
* also add unit test for this.

* change std::vector<int> to const reference

* Add const to GradientMachine::eval

* A tiny fix in PyDataProvider2

* hidden decorator kwargs in DataProvider.__init__
* also add unit test for this.

* Also add const to makeEvaluator

* Change member variables from public to protected

* Expose Evaluator API

* Complete a very simple mnist demo.

* Fix typo in PyDataProvider2.py

* Fix data provider bug in srl demo

Once encoutering a single sample whose size is larger than batch size,
the provider will yield empty batch and terminate the current pass unexpectedly
if can_over_batch_size=False.

* Remove utils/CommandLineParser.h

* Change float to real in NormLayer.h

* Add AverageOptimizer, Add save parameter

* Update index_en.md

删除了关于train.sh文件路径的说明

* Update index_cn.md

按照review建议进行了修改

* Add gitignore

* fix conflicts not found by git diff

* Fix gflags 2.1 issue

* Always use copy method for numpy.

* Make this demo support GPU

* Test GPU

* Test on GPU

* Remove bazel from dockerfile and docs

* Add comments.

* Remove all bazel tests in third party

* Modified VM OS from Debian to CoreOS in doc.

* add translation for cmd_parameter

* Bug fix for BatchNorm when testing more than models in test mode.

* add config_parser in trainer_config_helpers to seperate trainer config

* add config_parser_utils

* Remove unnecessary import in api_train.py

* enable clang-format

* modify documents

* modify documents

* follow comments: remove flags control

* move TestMain.cpp into utils and fix some tests for check

* fix some tests which only need paddle::initMain in main

* Add c++11 build python binding package

* Add common.h and remove DisableCopy and Typedefs

* follow some comments with issue 973

* follow some comments whit pr 854 after committed

* Update arguments_cn.md

* Update detail_introduction_cn.md

* Update detail_introduction_en.md

* add prefetch and onPassEnd to PaddleApi.h

* fix bug:
 * gradient_clipping_threshold should be allowed to set with parameter-grain

* add cluster train for quick_start

* Rename math.py to layer_math.py

* Fix #903

* Add externel google's dependencies

* Add other extern dependencies

* add remote updater in api and swig

* Fix merge errors.

* Update index_cn.rst

* Update external libs

* Refine cmake file names

* Remove paddle internals

* Refine CMakeLists

* Update external dependencies cmake

* Fix glog check type unmatch

* Add external_project_dependencies for targets

* Fix external denpendencies build order

* do not init localUpdater

* Fix warpctc header directory

* Clean Travis ci and fix bug

* Check md5 when download word embedding.

Fix #1008

* udw nullptr as localUpdater

* add new_layer_cn doc

* fix format problem in new_layer_cn.rst

* move TestMain.cpp to testing

* fix error when WITH_TESTING=OFF

* remove gtest_main_library

* fix link order error

* add python version evaluate function

* change LOG to log

* modify follow comment

* add first paddle function example for ContextProjectionForward operator,
by going through Daoyuan's excellent paddle function design.

* add ContextProjectionBackward, ContextProjectionBackwardData, ContextProjectionBackwardWeightw

* add Context Projection functions to Layer's forward and backward,
resolve merge conflicts

* combine data/weight to ContextProjectionBackward for clean code

* update interface of context projection functions, Tensor -> Matrix/Vector

* clean up unused code.

* Also check baidu.dict md5

* refine print information

* Add jupyter notebook support in Docker images.

* add translated chinese docs into catalog

* add paddle_test_util static lib to simplify unit test.

* add TestUtil.h and TestUtil.cpp, moving from gserver/tests/ to testing/

* Change run_all => entrypoint.

* fix dead links, reduce image size

* revise new_layer_cn and fix a few problems in new_layer_en

* change format

* a missing character in line 32

* Add extern python interp

* Add system configure

* Update external dependencies

* clean travis ci

* Clean cmake

* Fix merge conflict bug and glog

* Add python in travis ci for paddle version command

* Add wheel for paddle version command

* remove double spaces, remove chinese character "，"

* Update travis ci pip

* Move Execepts into arch/osx dir

* Revise common to Common

* Clean travis ci

* Add find system's swig
